libc::futimens is not available on Android

This function is used here: https://github.com/alexcrichton/filetime/blob/487c47d0a5f287564da8e9acd1cd5dd5a5f7869c/src/unix/utimensat.rs#L26

But it is not available on Android: https://stackoverflow.com/questions/19374749/how-to-work-around-absence-of-futimes-in-android-ndk

As a result, `ndk-make` throws an error `undefined reference to 'futimens'` when trying to build https://github.com/deltachat/deltachat-core-rust/pull/1749 which depends on [async-tar](https://lib.rs/crates/async-tar) which in turn depends on [filetime](https://lib.rs/crates/filetime).

--- a/src/unix/utimensat.rs
+++ b/src/unix/utimensat.rs
@@ -23,7 +23,7 @@ pub fn set_file_handle_times(
     mtime: Option<FileTime>,
 ) -> io::Result<()> {
     let times = [super::to_timespec(&atime), super::to_timespec(&mtime)];
-    let rc = unsafe { libc::futimens(f.as_raw_fd(), times.as_ptr()) };
+    let rc = unsafe { libc::utimensat(f.as_raw_fd(), core::ptr::null(), times.as_ptr(), 0) };
     if rc == 0 {
         Ok(())
     } else {

